src/main.cpp

```diff
@@ -13,55 +13,219 @@
 #include "titanfall.hpp"
 #include "titanfall2.hpp"
 #include "tricoll.hpp"
-
-
+#include "filesystem.h"
+#include <filesystem>
+#include <iostream>
+#include "Windows.h"
 #define PI 3.1415926536f
 typedef const char ModelDictEntry[128];
-
+namespace fs = std::filesystem;
 
 void print_usage(char* argv0) {
-    printf("USAGE: %s titanfall.bsp titanfall2.bsp\n", argv0);
-    // printf("USAGE: %s -d titanfall_dir/ titanfall2_dir/\n", argv0);
+    printf("USAGE: %s <game_dir> <titanfall.bsp> <output_folder>\n", argv0);
+}
+
+
+fs::path get_executable_path() {
+    char path[MAX_PATH];
+    GetModuleFileNameA(NULL, path, MAX_PATH);
+    return fs::path(path).parent_path();
+}
+void copy_model_files(const char* game_dir, const char* output_folder, const std::vector<std::string>& model_paths, const std::string& map_name) {
```

Comment:
I like how this function is structured with the comment labelled blocks
especially appreciate the use of `std::filesystem` to weld paths together